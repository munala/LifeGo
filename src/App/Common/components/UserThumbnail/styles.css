.user-thumbnail {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  font-family: sans-serif;
  font-size: 14px;
  padding: 10px 20px;
  cursor: pointer;
  background-color: #fff;
  border-bottom: 1px solid #e9e9e9;
}

.user-thumbnail:last-child {
  border-bottom: none;
}


.user-actions {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items:flex-start;
  margin-left: 10px;
}
.username {
  margin-left: 10px;
  color: #555;
  font-weight: bold;
}

.user-display-name {
  display: flex;
  flex-direction: row;
  text-decoration: none;
  cursor: pointer;
  line-height: 20px;
  height: 20px;
  vertical-align: baseline;
}

.user-alert-text {
  color: #777;
}

.user-alert-buttons {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
