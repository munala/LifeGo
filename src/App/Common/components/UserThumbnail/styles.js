export default {
  readBackground: {
    backgroundColor: '#f0f0f0',
  },
  ignore: {
    color: 'grey',
    backgroundColor: '#eee',
  },
  notificationAvatar: {
    display: 'flex',
    height: 60,
    width: 60,
  },
};
