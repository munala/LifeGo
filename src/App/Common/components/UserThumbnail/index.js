import React from 'react';
import PropTypes from 'prop-types';

import Avatar from '../Avatar';
import OutlinedButton from '../OutlinedButton';
import FlatButton from '../FlatButton';
import RaisedButton from '../RaisedButton';
import avatar from '../../../../assets/images/user.png';
import './styles.css';

const isFriend = ({ person, profile: { friends } }) => friends
  .some(friend => friend.id === person.id && person.id);

const UserThumbnail = ({
  profile,
  person,
  addFriend,
  removeFriend,
  deleteAlert,
  goToProfile,
  alert,
}) => {
  const isaFriend = isFriend({ profile, person });

  return (
    <div
      className="user-thumbnail"
      style={alert.read !== true ? { backgroundColor: '#f0f0f0' } : {}}
    >
      <Avatar
        style={{ display: 'flex', height: 60, width: 60 }}
        src={person.pictureUrl || avatar}
        onClick={goToProfile}
      />
      <div className="user-actions">
        <div
          className="user-alert-text"
          onClick={goToProfile}
        >
          <span className="username">{`${person.displayName} `}</span>
          <span className="alert-text">added you.</span>
        </div>
        <div className="user-alert-buttons">
          {
            isaFriend ?
              <OutlinedButton
                onClick={removeFriend}
                label="Remove"
              /> :
              <RaisedButton
                onClick={addFriend}
                label="Add"
              />
          }
          <FlatButton
            onClick={() => deleteAlert(alert)}
            style={{ color: 'grey', backgroundColor: '#eee' }}
            label="Ignore"
          />
        </div>
      </div>
    </div>
  );
};

UserThumbnail.propTypes = {
  profile: PropTypes.shape({
    id: PropTypes.number.isRequired,
    friends: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
    })).isRequired,
  }).isRequired,
  alert: PropTypes.shape({
    id: PropTypes.number,
    read: PropTypes.bool,
    user: PropTypes.string,
    userPictureUrl: PropTypes.string,
  }).isRequired,
  person: PropTypes.shape({
    id: PropTypes.number.isRequired,
  }).isRequired,
  addFriend: PropTypes.func.isRequired,
  removeFriend: PropTypes.func.isRequired,
  deleteAlert: PropTypes.func.isRequired,
  goToProfile: PropTypes.func.isRequired,
};

export default UserThumbnail;
