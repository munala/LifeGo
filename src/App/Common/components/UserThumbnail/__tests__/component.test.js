import React from 'react';
import { shallow } from 'enzyme';
import UserThumbnail from '../';

describe('UserThumbnail tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<UserThumbnail
      profile={{ id: 1, friends: [] }}
      person={{ id: 1, friends: [] }}
      addFriend={() => {}}
      removeFriend={() => {}}
      goToProfile={() => {}}
      alert={{}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
