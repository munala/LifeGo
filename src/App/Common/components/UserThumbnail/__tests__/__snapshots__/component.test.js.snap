// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserThumbnail tests renders without crashing 1`] = `
<div
  className="user-thumbnail"
  style={
    Object {
      "backgroundColor": "#f0f0f0",
    }
  }
>
  <UserAvatar
    name="user avatar"
    onClick={[Function]}
    src="user.png"
    style={
      Object {
        "display": "flex",
        "height": 60,
        "width": 60,
      }
    }
  />
  <div
    className="user-actions"
  >
    <div
      className="user-alert-text"
      onClick={[Function]}
    >
      <span
        className="username"
      >
        undefined 
      </span>
      <span
        className="alert-text"
      >
        added you.
      </span>
    </div>
    <div
      className="user-alert-buttons"
    >
      <WithStyles(RaisedButton)
        label="Add"
        onClick={[Function]}
      />
      <WithStyles(FlatButton)
        label="Ignore"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#eee",
            "color": "grey",
          }
        }
      />
    </div>
  </div>
</div>
`;
